staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -464,7 +464,20 @@ func newETCD3Storage(c storagebackend.ConfigForResource, newFunc, newListFunc fu
 
 	versioner := storage.APIObjectVersioner{}
 	decoder := etcd3.NewDefaultDecoder(c.Codec, versioner)
-	return etcd3.New(client, c.Codec, newFunc, newListFunc, c.Prefix, resourcePrefix, c.GroupResource, transformer, c.LeaseManagerConfig, decoder, versioner), destroyFunc, nil
+	listErrAggregatorFactory := etcd3.DefaultListErrorAggregatorFactory
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AllowUnsafeMalformedObjectDeletion) {
+		transformer = etcd3.WithCorruptObjErrorHandlingTransformer(transformer)
+		decoder = etcd3.WithCorruptObjErrorHandlingDecoder(decoder)
+		// TODO: make it configurable?
+		listErrAggregatorFactory = etcd3.CorruptObjErrAggregatorFactory(100)
+	}
+
+	store := etcd3.New(client, c.Codec, newFunc, newListFunc, c.Prefix, resourcePrefix, c.GroupResource, transformer, c.LeaseManagerConfig, decoder, versioner, listErrAggregatorFactory)
```

Comment:
why take `listErrAggregatorFactory` instead of setting it in `etcd3.newStore` based on the featuregate?

Reply:
all the initialization bits - wrapping the decoder and the transformer are also here, inside the factory.

Reply:
yes, done